js/src/contract/item.ts

```diff
@@ -62,23 +60,23 @@ function convertNumsToBNs(data: any) {
         k.callback.code = new BN(k.callback.code);
       }
 
-      let u = k.itemClassType.consumable;
+      let c = k.itemClassType.consumable;
 
-      if (u) {
-        if (u.maxUses != null) u.maxUses = new BN(u.maxUses);
-        if (u.maxPlayersPerUse != null)
-          u.maxPlayersPerUse = new BN(u.maxPlayersPerUse);
-        if (u.warmupDuration != null)
-          u.warmupDuration = new BN(u.warmupDuration);
-        if (u.cooldownDuration != null) {
-          u.cooldownDuration = new BN(u.cooldownDuration);
+      if (c) {
+        if (c.maxUses != null) c.maxUses = new BN(c.maxUses);
+        if (c.maxPlayersPerUse != null)
+          c.maxPlayersPerUse = new BN(c.maxPlayersPerUse);
+        if (c.warmupDuration != null)
+          c.warmupDuration = new BN(c.warmupDuration);
+        if (c.cooldownDuration != null) {
+          c.cooldownDuration = new BN(c.cooldownDuration);
         }
       }
 
-      u = k.itemClassType.wearable;
-      if (u) {
-        if (u.limitPerPart) {
-          u.limitPerPart = new BN(u.limitPerPart);
+      let w = k.itemClassType.wearable;
+      if (w) {
+        if (w.limitPerPart) { // QUESTION: This will not be converted if 0 - is that intended?
```

Comment:
That is not intended. It should be checking for existence, not just truthiness. There is a helper function in sol-kit that does this type of conversion https://github.com/raindrop-studios/sol-kit/blob/4c629bb8f14f215317d01716b6dbb8855b7eb443/src/instruction/utils.ts#L3

Here is a snippet of it being used with a relatively complex data structure

```
InstructionUtils.convertNumbersToBNs(args, [
      "bump",
      "rainTreasuryDenominator",
      "gamesConfig.map.grid.maxX",
      "gamesConfig.map.grid.maxY",
      "gamesConfig.explore.baseReward",
      "gamesConfig.explore.levels.[].count",
      "gamesConfig.explore.levels.[].price",
      "gamesConfig.explore.levels.[].rewardMultiplier",
      "gamesConfig.stack.baseReward",
      "gamesConfig.stack.levels.[].count",
      "gamesConfig.stack.levels.[].price",
      "gamesConfig.stack.levels.[].rewardMultiplier",
    ]);
```